File: Wms.Api/Dto/Expense/ExpenseDetails/ExpenseDetailsDto.cs
using System.ComponentModel.DataAnnotations;

namespace Wms.Api.Dto.Expense.ExpenseDetails
{
    public class ExpenseDetailsDto
    {
        public Guid Id { get; set; }
        public string Description { get; set; } = default!;
        public decimal Amount { get; set; }
        public string Category { get; set; } = default!;  // Freight, Internet, Staff Refreshment, etc.
        public string? Remark { get; set; }

        public DateTime CreatedAt { get; set; }
        public Guid CreatedById { get; set; }
        public DateTime? ChangedAt { get; set; }
        public Guid? ChangedById { get; set; }
    }

    public class ExpenseFindByParametersDto : PagedRequestAbstractDto
    {
        public Guid[] ExpenseIds { get; set; } = [];
    }
}

File: Frontend/src/interfaces/v12/expense/expenseDetails/expenseDetailsDto.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";

export interface ExpenseDetailsDto extends CreatedChangedEntity {
  id: guid;
  description: string;
  amount: number;
  category: string; // Freight, Internet, Staff Refreshment, etc.
  remark?: string;
}

export interface ExpenseFindByParametersDto {
  expenseIds: guid[];
  page: number;
  pageSize: number;
}
